### PR TITLE
Migrate Request/Response/Result classes

### PR DESCRIPTION
**Pull request by rpdome**
This is a straightforward PR, moving classes from common to common4j. 

The code are barely modified (except a couple spots that was reported by spotbugs).

Broker PR: https://github.com/AzureAD/ad-accounts-for-android/pull/1601
MSAL PR: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1444
ADAL PR: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1615

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftTokenResponse.java [hunk 1]
```diff
@@ -261,7 +261,11 @@ public void setClientInfo(final String clientInfo) {
      * @return The ext expires on to get.
      */
     public Date getExtExpiresOn() {
-        return mExtExpiresOn;
+        if (mExtExpiresOn == null){
```
Comment (rpdome) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftTokenResponse.java:264]: Here (and 277) are the only changes I made (for spotbugs).

Comment (shahzaibj) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftTokenResponse.java:264]:
```suggestion
        if (mExtExpiresOn == null) {
```

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsTokenResponse.java [hunk 2]
```diff
@@ -52,4 +52,4 @@ public void setExpiresNotBefore(final String expiresNotBefore) {
 
 
 
-}
+}
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsTokenResponse.java:55]: new line ending

File: common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationResponse.java [hunk 3]
```diff
@@ -135,4 +135,4 @@ public void setExtraParameters(Iterable<Map.Entry<String, String>> params) {
         public static final String CODE = "code";
         public static final String STATE = "state";
     }
-}
+}
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationResponse.java:138]: new line ending

**Review by shahzaibj — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsTokenRequest.java [hunk 4]
```diff
@@ -1,9 +0,0 @@
-package com.microsoft.identity.common.internal.providers.microsoft.microsoftsts;
-
-import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftTokenRequest;
-
-public class MicrosoftStsTokenRequest extends MicrosoftTokenRequest {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsTokenRequest.java:5]: Is this safe?

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsTokenRequest.java:5]: Yes.

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftTokenResponse.java [hunk 5]
```diff
@@ -270,7 +274,11 @@ public Date getExtExpiresOn() {
      * @param extExpiresOn The expires on to set.
      */
     public void setExtExpiresOn(final Date extExpiresOn) {
-        mExtExpiresOn = extExpiresOn;
+        if (extExpiresOn == null) {
```
Comment (AdamBJohnsonx) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftTokenResponse.java:277]: should we set mExtExpiresOn to null in this case?

**Review by AdamBJohnsonx — APPROVED** [review]
A couple of questions - we should be consistent and clear about the use or not of null in the one class.

**Pull request merged**